plugins/inputs/docker/docker_test.go

```diff
@@ -1117,3 +1125,191 @@ func TestHostnameFromID(t *testing.T) {
 	}
 
 }
+
+func Test_parseContainerStatsPerDeviceAndTotal(t *testing.T) {
+	type args struct {
+		stat             *types.StatsJSON
+		acc              telegraf.Accumulator
+		tags             map[string]string
+		id               string
+		perDeviceInclude []string
+		totalInclude     []string
+		daemonOSType     string
+	}
+	stats := testStats()
+	tests := []struct {
+		name string
+		args args
+	}{
+		{
+			name: "Per device and total metrics enabled",
+			args: args{
+				stat:             stats,
+				perDeviceInclude: containerMetricClasses,
+				totalInclude:     containerMetricClasses,
+			},
+		},
+		{
+			"Per device metrics enabled",
+			args{
+				stat:             stats,
+				perDeviceInclude: containerMetricClasses,
+				totalInclude:     []string{},
+			},
+		},
+		{
+			"Total metrics enabled",
+			args{
+				stat:             stats,
+				perDeviceInclude: []string{},
+				totalInclude:     containerMetricClasses,
+			},
+		},
+		{
+			name: "Per device and total metrics disabled",
+			args: args{
+				stat:             stats,
+				perDeviceInclude: []string{},
+				totalInclude:     []string{},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var acc testutil.Accumulator
+			parseContainerStats(tt.args.stat, &acc, tt.args.tags, tt.args.id, tt.args.perDeviceInclude,
+				tt.args.totalInclude, tt.args.daemonOSType)
+
+			var numCpus int
+			if stats.CPUStats.OnlineCPUs > 0 {
+				numCpus = len(stats.CPUStats.CPUUsage.PercpuUsage[:stats.CPUStats.OnlineCPUs])
+			} else {
+				numCpus = len(stats.CPUStats.CPUUsage.PercpuUsage)
+			}
+
+			numNetworks := len(stats.Networks)
+
+			deviceStatMap := getDeviceStatMap(stats.BlkioStats)
+
+			// per device
+			for i := 0; i < numCpus; i++ {
+				if choice.Contains("cpu", tt.args.perDeviceInclude) {
+					acc.AssertTagValuesInMeasurement(t, "docker_container_cpu",
+						map[string]string{"cpu": fmt.Sprintf("cpu%d", i)})
+				} else {
+					acc.AssertTagValuesNotInMeasurement(t, "docker_container_cpu",
+						map[string]string{"cpu": fmt.Sprintf("cpu%d", i)})
```

Comment:
This is probably very specific to your plugin. I suggest to include this test function here, using the `choice` option for doing the work.